packages/text-input/src/TextInput.tsx

```diff
@@ -119,10 +119,15 @@ const TextInput: TextInput = forwardRef<HTMLInputElement, TextInputProps>(
       );
     }
 
-    if (type === 'search' && !rest['aria-label']) {
-      console.error(
-        'For screen-reader accessibility, aria-label must be provided to TextInput.',
+    if (type === 'search') {
+      console.warn(
```

Comment:
Should consider using `lib/consoleOnce`, otherwise this will warn on every render